### PR TITLE
[Merged by Bors] - feat: add clear! tactic

### PR DESCRIPTION
**Pull request by JOSHCLUNE**

### CODE REVIEW

File: Mathlib/Tactic/Clear!.lean [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    let fvarIds ← getFVarIds hs
+    liftMetaTactic1 fun goal => do
+      let mut toClear : Array FVarId := #[]
+      for decl in ← getLCtx do
+        let hasAnyDependencyFn (fvarId : FVarId) (acc : Bool) : MetaM Bool :=
+          return acc || decl.fvarId == fvarId || (← localDeclDependsOn decl fvarId)
+        let isDependent ← Array.foldrM hasAnyDependencyFn false fvarIds
+        if(isDependent) then
+          if let none ← isClass? decl.type then
+            toClear := toClear.push decl.fvarId
```
Comment (digama0) [inline Mathlib/Tactic/Clear!.lean:27]:
```suggestion
      let lctx ← getLCtx
      for fvar in fvarIds do
        let deps := (← collectForwardDeps #[mkFVar fvar] true).map (·.fvarId!)
        if ← deps.allM fun dep => return (← isClass? (lctx.get! dep).type).isNone then
          toClear := toClear ++ deps
```

**Comment by semorrison** [comment]
@JOSHCLUNE, it's helpful after you make changes suggested by reviews if you remove the `awaiting-author` label, and add the `awaiting-review` label. Otherwise people may not see that the PR needs attention again!

**Comment by JOSHCLUNE** [comment]
> @JOSHCLUNE, it's helpful after you make changes suggested by reviews if you remove the `awaiting-author` label, and add the `awaiting-review` label. Otherwise people may not see that the PR needs attention again!

Yes, sorry about that. Unfortunately, as best as I can tell, I don't appear to have the permissions to do that (I'm not a regular mathlib contributor so I don't have write access to the repo). But I'll double check today with others that are more Github-savvy to confirm that I'm not simply overlooking the way to do that.

**Comment by semorrison** [comment]
Sorry about that @JOSHCLUNE, I'd misunderstood how permissions worked. I've given you access to push branches, and modify labels.

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**